api/proto/external.proto

```diff
@@ -267,3 +267,28 @@ message Receipt {
     // Note: This value is self-reported by the sender and is unverifiable.
     Amount amount = 4;
 }
+
+///////////////////////////////////////////////////////////////////////////////
+// `watcher/api` crate
```

Comment:
Is it worth putting this in a separate `api/proto/watcher.proto` file?

Reply:
I don't have a strong preference. Does anyone else does? I believe the idea of `external.proto` is to collect all the proto stuff we want to expose to non-Rust clients.

Reply:
The distinction I was making in my head is that the watcher isn't part of the "core" mobilecoin, similar to how `PrintableWrapper` is in it's own proto file. I'm not sure if there's a benefit to having *all* external types in a single proto file or not? Maybe `external.proto` should really be called something like `core.proto`?

Reply:
Okay, I'll move this into `watcher.proto`. Everything else is definitely out of scope for this PR.